src/js_realm.hpp

```diff
@@ -67,7 +67,24 @@ class RealmClass;
 
 template<typename T>
 class RealmDelegate : public BindingContext {
-  public:
+private:
+    void did_change(std::vector<ObserverState> const&, std::vector<void*> const&, bool) override {
+        HANDLESCOPE
```

Comment:
Why do you need `HANDLESCOPE`?

Reply:
notify() allocates a JS object, so node requires an explicit scope for our local reference. Putting the HANDLESCOPE inside notify() doesn't work for the schema change case.